src/transformers/models/falcon/modeling_falcon.py

```diff
@@ -129,6 +129,11 @@ def cos_sin(
         total_length = seq_len + past_key_values_length
         if total_length > self.seq_len_cached:
             self._set_cos_sin_cache(total_length, device, dtype)
+
+        # the cached tensors need to update their devices (for example, after we change the model's device)
+        self.cos_cached = self.cos_cached.to(device)
+        self.sin_cached = self.sin_cached.to(device)
```

Comment:
I removed the check of device. As we can't compare device (given by a tensor) with a string directly.
If it is on the same device (inferred by torch itself), there won't be any operation performed.